### PR TITLE
Do not fail on query errors

### PR DESCRIPTION
**Pull request by mattflax**
The current process will die if a query cannot be parsed for any reason. If running a large set of queries from another source, this can result in a lengthy debugging process for the queries, or make them less precise by running generic regular expressions to substitute illegal characters.

This patch catches Solr and Elasticsearch exceptions when running the query, logs them, and returns an empty query result rather than throwing a RuntimeException and making the Engine crash.

### CODE REVIEW

**Comment by agazzarini** [comment]
Hi @mattflax, thank you very much for your contributions. This week is quite a nightmare in terms of commitments: I promise I will have a look as soon as I can 

Thanks again, 
Andrea

**Comment by mattflax** [comment]
No worries - thanks for letting me know. Our current project is likely to generate a few pull requests, so sorry in advance!

File: rre-search-platform/rre-search-platform-elastic-search-impl/src/main/java/io/sease/rre/search/api/impl/Elasticsearch.java [hunk 1]
```diff
@@ -199,7 +201,12 @@ public QueryOrSearchResponse executeQuery(final String indexName, final String q
                                 return result;
                             })
                             .collect(toList()));
-        } catch (final IOException exception) {
+        }
+        catch (ElasticsearchException e) {
```
Comment (agazzarini) [inline rre-search-platform/rre-search-platform-elastic-search-impl/src/main/java/io/sease/rre/search/api/impl/Elasticsearch.java]: Just a minor thing @mattoscln @mattflax could you please put the catch in the same line of }?
I know, that's a matter of style, no good or bad, but I think we should follow the same style as much as possible ok?

File: rre-search-platform/rre-search-platform-elastic-search-impl/src/main/java/io/sease/rre/search/api/impl/Elasticsearch.java [hunk 2]
```diff
@@ -199,7 +201,12 @@ public QueryOrSearchResponse executeQuery(final String indexName, final String q
...
+            LOGGER.error("Caught ElasticsearchException :: " + e.getMessage());
+            return new QueryOrSearchResponse(0, Collections.emptyList());
+        }
+        catch (final IOException exception) {
```
Comment (agazzarini) [inline rre-search-platform/rre-search-platform-elastic-search-impl/src/main/java/io/sease/rre/search/api/impl/Elasticsearch.java]: same thing as above

File: rre-search-platform/rre-search-platform-solr-impl/src/main/java/io/sease/rre/search/api/impl/ApacheSolr.java [hunk 3]
```diff
@@ -138,7 +143,12 @@ public QueryOrSearchResponse executeQuery(final String coreName, final String qu
                                     response.getResults().getNumFound(),
                                     new ArrayList<Map<String, Object>>(response.getResults())))
                     .get();
-        } catch (final Exception exception) {
+        }
+        catch (SolrException e) {
```
Comment (agazzarini) [inline rre-search-platform/rre-search-platform-solr-impl/src/main/java/io/sease/rre/search/api/impl/ApacheSolr.java]: see the formatting comments in the ElasticSearch class

File: rre-search-platform/rre-search-platform-solr-impl/src/main/java/io/sease/rre/search/api/impl/ApacheSolr.java [hunk 4]
```diff
@@ -138,7 +143,12 @@ public QueryOrSearchResponse executeQuery(final String coreName, final String qu
...
+            LOGGER.error("Caught Solr exception :: " + e.getMessage());
+            return new QueryOrSearchResponse(0, Collections.emptyList());
+        }
+        catch (final Exception exception) {
```
Comment (agazzarini) [inline rre-search-platform/rre-search-platform-solr-impl/src/main/java/io/sease/rre/search/api/impl/ApacheSolr.java]: see above

**Comment by agazzarini** [comment]
LGTM: just few minor formatting things

File: rre-search-platform/rre-search-platform-elastic-search-impl/src/main/java/io/sease/rre/search/api/impl/Elasticsearch.java (hunk 1 above)
Comment (mattflax) [reply rre-search-platform/rre-search-platform-elastic-search-impl/src/main/java/io/sease/rre/search/api/impl/Elasticsearch.java]: Thanks for this - I should have checked the formatting before committing. I've pushed changes to this and the other formatting issues.

**Pull request merged**